### DISCUSSION TITLE
Documentation improvements

### DISCUSSION THREAD

**Discussion by UPLukeet** [surevine/govuk-react-jsx#121] [Ideas]
I think the install instructions could be much clearer for instance the fact you need to import the gov css file is never mentioned. I also think interaction with components need to take props like a standard react component `<Input value={} onChange={} />` or the way you interact with them currently should be better written up. For instance storybook should list all the props for each element and what values they accept(string, boolean, etc).

**Comment by andymantell** [comment]
![image](https://user-images.githubusercontent.com/134642/136521866-ac3b857a-f0f9-459b-a7f3-bfa1e210552d.png)

The Gov CSS is mentioned under assumptions, but I agree it could probably do with being more explicit in the install section. I imagine many won't read those assumptions. Good shout.

As for the way the props are received - a bit more documentation here sounds good, I agree. The example you give should work though - the input component can take value and onChange props.

**Comment by andymantell** [comment]
Kinda mirrored by this too:

https://github.com/surevine/govuk-react-jsx/issues/47

The current storybook demos are automatically generated from the govuk-frontend examples, but they're not very useful from a documentation perspective.

**Comment by merryfaceTech** [comment]
I'd like to echo some of the sentiment here. The lack of documentation is making it quite difficult to work with this. Certain things behave in mysterious ways and having to jump into the node modules to decipher what was written can be tedious and difficult at times. Some form of documentation would be tremendously helpful

**Comment by andymantell** [comment]
I'll do my best when I can, but this package is currently maintained at the weekend so it's difficult to prioritise this vs spending time with my family. Very happy to accept contributions. Documentation is a great way to give back :-)

**Reply by merryfaceTech** [reply]
I'd happily try to do some of it, but I'm quite a junior developer and don't feel like I could do the docs justice. I would write it in some simple, practical, and probably not technically accurate, prose.
